src/components/fileviewer.js

```diff
@@ -189,37 +187,12 @@ export default class FileViewerContainer extends Component {
       );
     };
 
-    /* This component contains metadata of the file */
-    const FileViewerMeta = ({self}) => {
-      console.log("run render");
-      return (
-        <div className="file-meta-viewer">
-          <div className="file-meta-center">
-            <div className="file-meta-title">File Coverage</div>
-            <CoveragePercentageViewer
-              coverage={self.state.coverage}
-            />
-            <FileViewerStatus ref={
-              c => {
-                console.log("assign:" +c);
-                if (c) self.status = c
-              }
-            }/>
-          </div>
-          {self.status.app && <span className="error_message">{self.status.app}</span>}
-
-          <div className="file-summary">
-            <div className="file-path">{self.path}</div>
-          </div>
-          <div className="file-meta-revision">revision number: {self.revision}</div>
-        </div>
-      );
-    };
-
```

Comment:
technically, `render()` is run multiple times, but only things that change are rerun.  In the case of `FileViewerMeta` (as an inner function of `render()`) It is a function, but a new function frame is created each time `render()` is run.  This makes a new instance of `FileViewerMeta`, which React detects; so it runs the vitual `render()`  for his new frame, which creates a new instance of `FileViewerStatus`.  React will detach the old `FileViewerStatus` instance and attach the new. 

`FileViewerMeta` must stay out of the `render()` function so it has only one frame, runs `render()` just once, and creates only one `FileViewerStatus`.  this means multiple constrictor variables must be passed in (or just all of `self=this`)